### ISSUE TITLE
Add neptune.ai for metadata collection of model training

### ISSUE THREAD

**Issue by abuzarmahmood** [pulakatlab/auto_slide#15]

**Comment by abuzarmahmood** [comment]
### Summary of Suggestions and Changes

1. **Files Identified for Modification:**
   - **File**: `/home/exouser/Desktop/blech_github_bot/src/repos/pulakatlab/auto_slide/src/pipeline/model/training.py`
     - **Description**: Contains the main logic for model training. Relevant for integrating `neptune.ai` to track and log metadata during the training process.
   - **File**: `/home/exouser/Desktop/blech_github_bot/src/repos/pulakatlab/auto_slide/src/pipeline/run_pipeline.py`
     - **Description**: Orchestrates the complete AutoSlide pipeline, including model training. Modifications here are necessary to initialize and configure `neptune.ai` tracking for the training step.

2. **Plan to Address the Issues:**
   - **Initialize Neptune.ai**: Add initialization code for `neptune.ai` in the `run_pipeline.py` file to set up the tracking environment.
   - **Log Metadata**: Modify the `training.py` file to log relevant training metrics and parameters to `neptune.ai` during the training loop.

3. **Specific Changes Suggested:**

   1. **Initialize Neptune.ai**
      - **File**: `/home/exouser/Desktop/blech_github_bot/src/repos/pulakatlab/auto_slide/src/pipeline/run_pipeline.py`
      - **Line**: Before line 113 (before the `main()` function)
      - **Description**: Add code to initialize `neptune.ai` with the appropriate API token and project name.
      - **Code Snippet**:
        ```python
        import neptune.new as neptune

        # Initialize Neptune
        run = neptune.init(
            project='your_workspace/your_project',
            api_token='your_api_token',  # replace with your Neptune API token
        )
        ```

   2. **Log Metadata During Training**
      - **File**: `/home/exouser/Desktop/blech_github_bot/src/repos/pulakatlab/auto_slide/src/pipeline/model/training.py`
      - **Line**: Around line 257 (inside the training loop)
      - **Description**: Log training loss and validation loss to `neptune.ai` at each epoch.
      - **Code Snippet**:
        ```python
        for epoch in trange(n_epochs):
            # Existing training code...
            
            # Log training and validation loss to Neptune
            run['train/loss'].log(train_epoch_loss)
            run['val/loss'].log(val_epoch_loss)
            
            # Existing code for saving best model...
        ```

   3. **Close Neptune Run**
      - **File**: `/home/exouser/Desktop/blech_github_bot/src/repos/pulakatlab/auto_slide/src/pipeline/run_pipeline.py`
      - **Line**: After line 141 (after the pipeline completion)
      - **Description**: Ensure the Neptune run is closed after the pipeline execution.
      - **Code Snippet**:
        ```python
        # Close the Neptune run
        run.stop()
        ```

These changes will integrate `neptune.ai` into the model training process, allowing for comprehensive tracking and logging of training metadata. Make sure to replace `'your_workspace/your_project'` and `'your_api_token'` with actual values from your Neptune.ai account.

---
*This response was automatically generated by blech_bot using model gpt-4o*

---
*This response was automatically generated by blech_bot*
